cmd/tink-cli/cmd/hardware/mac.go

```diff
@@ -28,20 +31,44 @@ var macCmd = &cobra.Command{
 		return nil
 	},
 	Run: func(cmd *cobra.Command, args []string) {
+		t := table.NewWriter()
+		t.SetOutputMirror(os.Stdout)
+		t.AppendHeader(table.Row{"Field Name", "Value"})
+
 		for _, mac := range args {
 			hw, err := client.HardwareClient.ByMAC(context.Background(), &hardware.GetRequest{Mac: mac})
 			if err != nil {
 				log.Fatal(err)
 			}
-			b, err := json.Marshal(hw)
-			if err != nil {
-				log.Fatal(err)
+			if hw.GetId() == "" {
+				log.Fatal("Mac address not found in the database ", mac)
 			}
-			fmt.Println(string(b))
+			if !mData {
+				interfaces := hw.GetNetwork().GetInterfaces()
+				for i := range hw.Network.Interfaces {
+					if hw.Network.Interfaces[i].Dhcp.Mac == mac {
+						t.AppendRow(table.Row{"MAC Address", interfaces[i].Dhcp.Mac})
+						t.AppendRow(table.Row{"Hardware-Id", hw.Id})
+						t.AppendRow(table.Row{"IP Address", interfaces[i].Dhcp.Ip.Address})
+						t.AppendRow(table.Row{"Hostname", interfaces[i].Dhcp.Hostname})
+					}
+				}
+			} else {
+				fmt.Println(hw)
+			}
+		}
+		if !mData {
+			t.Render()
 		}
 	},
 }
 
+func addMacFlags() {
+	flags := macCmd.Flags()
+	flags.BoolVarP(&mData, "details", "d", false, "Provide the complete hardware details in json format")
```

Comment:
I wonder why it is called `mData` is this function and `ipData` in the other implementation? Is there a way this rendering logic could be shared between `ip` and `mac` subcommands, so they always behave the same? These two functions seem, to me, to be two ways of accessing the same data and I would expect would behave the same.

Reply:
Suggestion seems to be good and optimized. And I think it should work as well if the rendering logic happen in a function which should be called by each of the command. I'll work on it.